x/crosschain/keeper/in_tx_tracker.go

```diff
@@ -0,0 +1,185 @@
+	b := store.Get(types.KeyPrefix(key))
+	if b == nil {
+		return val, false
+	}
+	k.cdc.MustUnmarshal(b, &val)
+	return val, true
+}
+
+func (k Keeper) RemoveInTxTrackerIfExists(ctx sdk.Context, chainID int64, txHash string) {
+	key := getInTrackerKey(chainID, txHash)
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.InTxTrackerKeyPrefix))
+	if store.Has(types.KeyPrefix(key)) {
+		store.Delete(types.KeyPrefix(key))
+	}
+}
+func (k Keeper) GetAllInTxTracker(ctx sdk.Context) (list []types.InTxTracker) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.InTxTrackerKeyPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.InTxTracker
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+	return list
+}
+
+func (k Keeper) GetAllInTxTrackerForChain(ctx sdk.Context, chainID int64) (list []types.InTxTracker) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.InTxTrackerKeyPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte(fmt.Sprintf("%d-", chainID)))
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.InTxTracker
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+	return list
+}
+
+func (k Keeper) GetAllInTxTrackerForChainPaginated(ctx sdk.Context, chainID int64, pagination *query.PageRequest) (inTxTrackers []types.InTxTracker, pageRes *query.PageResponse, err error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(fmt.Sprintf("%s", types.InTxTrackerKeyPrefix)))
+	chainStore := prefix.NewStore(store, types.KeyPrefix(fmt.Sprintf("%d-", chainID)))
+	pageRes, err = query.Paginate(chainStore, pagination, func(key []byte, value []byte) error {
+		var inTxTracker types.InTxTracker
+		if err := k.cdc.Unmarshal(value, &inTxTracker); err != nil {
+			return err
+		}
+		inTxTrackers = append(inTxTrackers, inTxTracker)
+		return nil
+	})
+	return
+}
+
+func (k msgServer) AddToInTxTracker(goCtx context.Context, msg *types.MsgAddToInTxTracker) (*types.MsgAddToInTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observerTypes.ErrSupportedChains
+	}
+
+	adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observerTypes.Policy_Type_group1)
+	isAdmin := msg.Creator == adminPolicyAccount
+
+	isObserver, err := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+	if err != nil {
+		ctx.Logger().Error("Error while checking if the account is an observer", err)
+	}
+	isProven := false
+	if msg.Proof != nil {
+		isProven, err = k.VerifyInTxTrackerProof(ctx, msg.Proof, msg.BlockHash, msg.TxIndex, msg.ChainId, msg.CoinType)
+		if err != nil {
+			return nil, err
```

Comment:
We should use error wrapper in the message body for debugging purposes

Reply:
Just saw that we have the error wrapping in `VerifyInTxTrackerProof`. I'm thinking that maybe we could return the simple error in `VerifyInTxTrackerProof` and what wraps it here with  `types.ErrCannotVerifyProof` ?